contracts/traits/ChargesEthFee.sol

```diff
@@ -9,65 +9,24 @@
 
 pragma solidity ^0.5.13;
 
+import "../interfaces/IEthFeeRegistry.sol";
 
-contract ChargesEthFee {
-  event SetFeeManager(address addr);
-  event SetFeeCollector(address addr);
-  event SetEthFee(uint256 ethFee);
-  event WithdrawEth(address indexed to, uint256 amount);
-
-  uint256 public ethFee;
-
-  address public feeManager;
-  address public feeCollector;
-
-  modifier onlyFeeManager() {
-    require(msg.sender == feeManager, "ChargesEthFee: caller is not the feeManager");
-    _;
-  }
-
-  modifier onlyFeeCollector() {
-    require(msg.sender == feeCollector, "ChargesEthFee: caller is not the feeCollector");
-    _;
-  }
-
-  constructor() public {
-
-  }
-
-  // SETTERS
-
-  function setFeeManager(address _addr) external onlyFeeManager {
-    feeManager = _addr;
 
-    emit SetFeeManager(_addr);
-  }
-
-  function setFeeCollector(address _addr) external onlyFeeManager {
-    feeCollector = _addr;
-
-    emit SetFeeCollector(_addr);
-  }
-
-  function setEthFee(uint256 _ethFee) external onlyFeeManager {
-    ethFee = _ethFee;
-
-    emit SetEthFee(_ethFee);
-  }
-
-  // WITHDRAWERS
-
-  function withdrawEth(address payable _to) external onlyFeeCollector {
-    uint256 balance = address(this).balance;
+contract ChargesEthFee {
+  address internal _feeRegistry;
 
-    _to.transfer(balance);
+  constructor() public {}
 
-    emit WithdrawEth(_to, balance);
+  function feeRegistry() public returns(address) {
+    return _feeRegistry;
   }
 
   // INTERNAL
 
-  function _acceptPayment() internal {
-    require(msg.value == ethFee, "Fee and msg.value not equal");
+  function _acceptPayment(bytes32 _key) internal {
+    address payable feeRegistryPayable = address(uint160(feeRegistry()));
+
+    require(msg.value == IEthFeeRegistry(feeRegistryPayable).ethFeeByKey(_key), "Fee and msg.value not equal");
+    feeRegistryPayable.transfer(msg.value);
```

Comment:
If a `feeRegistry` is wrapped with proxy, make sure this transfer can pass through its fallback function.